rdt-open-reader/src/main/java/io/ona/rdt/util/RDTSyncConfiguration.java

```diff
@@ -28,16 +36,41 @@ public int getSyncMaxRetries() {
 
     @Override
     public SyncFilter getSyncFilterParam() {
-        return BuildConfig.SYNC_FILTER_PARAM.equals(Constants.Config.TEAM)
-                ? SyncFilter.TEAM_ID : SyncFilter.PROVIDER;
+        return BuildConfig.SYNC_FILTER_PARAM.equals(Constants.Config.LOCATION)
+                ? SyncFilter.LOCATION_ID : SyncFilter.PROVIDER;
     }
 
     @Override
     public String getSyncFilterValue() {
         AllSharedPreferences sharedPreferences = RDTApplication.getInstance().getContext().userService().getAllSharedPreferences();
         String provider = sharedPreferences.fetchRegisteredANM();
-        return BuildConfig.SYNC_FILTER_PARAM.equals(Constants.Config.TEAM)
-                ? sharedPreferences.fetchDefaultTeamId(provider) : provider;
+        if (BuildConfig.SYNC_FILTER_PARAM.equals(Constants.Config.LOCATION)) {
+            LocationServiceHelper locationServiceHelper = LocationServiceHelper.getInstance();
+
+            String locationId = sharedPreferences.fetchDefaultLocalityId(provider);
+            LocationTree locationTree = locationServiceHelper.getLocationHierarchy(locationId);
```

Comment:
This is already done here https://github.com/onaio/rdt-standard/blob/42662cc88d98c458249c2d3d3144f9c9a5454b7e/rdt-open-reader/src/main/java/io/ona/rdt/sync/RDTSyncIntentService.java#L40.

Reply:
You mean that I take those two out into to a new method?

Reply:
I mean that this sync of hierarchy is already done in the app. You probably only need to parse the location into csv and save that as a preference to be used in the `RDTSyncIntentService`.

Reply:
Cannot, its blocked by the if-block. `parentLocationId` always null for supervisor.

Reply:
remove the if block and set the locationid to the node itself if it doesn't have a parent

Reply:
also in this [method](https://github.com/onaio/rdt-standard/blob/42662cc88d98c458249c2d3d3144f9c9a5454b7e/rdt-open-reader/src/main/java/io/ona/rdt/sync/RDTSyncIntentService.java#L35), perform the extraction by calling [this](https://github.com/onaio/rdt-standard/blob/42662cc88d98c458249c2d3d3144f9c9a5454b7e/rdt-open-reader/src/covidIndonesia/java/io/ona/rdt/util/CovidRDTJsonFormUtils.java#L213) and save the locations to sharedpreferences as csv.

Reply:
then modify this [method](https://github.com/onaio/rdt-standard/blob/42662cc88d98c458249c2d3d3144f9c9a5454b7e/rdt-open-reader/src/covidIndonesia/java/io/ona/rdt/util/CovidRDTJsonFormUtils.java#L188) to simply ready from sharedpreferences and create a jsonarray from the csv.

Reply:
something to note is that the logic [here](https://github.com/onaio/rdt-standard/blob/42662cc88d98c458249c2d3d3144f9c9a5454b7e/rdt-open-reader/src/covidIndonesia/java/io/ona/rdt/util/CovidRDTJsonFormUtils.java#L188) extracts locations under the parent of the user's location (including itself).